Services/ProductService.cs

```diff
@@ -0,0 +1,52 @@
+﻿using EFCoreServices.DTO;
+using EFCoreServices.Models;
+using EFCoreServices.Repository;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace EFCoreServices.Services
+{
+    public class ProductService
+    {
+        private readonly IRepository<Product> _repository;
+
+        public ProductService(IRepository<Product> repository)
+        {
+            _repository = repository;
+        }
+
+        public Task<List<ProductDto>> GetProductsAsync() => _repository.GetProductsAsync();
+
+        public Task<ProductDto> GetByIDAsync(int productID) => _repository.GetByIDAsync(productID);
+
+        public async Task<ProductDto> AddAsync(ProductDto product)
+        {
+            var model = new Product()
+            {
+                Id = product.Id,
```

Comment:
id is the primary key it should be auto generated as identity not set when added